app/controllers/api/base_controller/renderer.rb

```diff
@@ -264,12 +264,29 @@ def fetch_direct_virtual_attribute(type, resource, attr)
         return unless attr_accessible?(resource, attr)
         virtattr_accessor = virtual_attribute_accessor(type, attr)
         value = virtattr_accessor ? send(virtattr_accessor, resource) : virtual_attribute_search(resource, attr)
-        value = add_custom_action_hrefs(value) if attr == "custom_actions"
+        value = handle_virtual_attributes(attr, value)
         result = {attr => normalize_attr(attr, value)}
         # set nil vtype above to "#{type}/#{resource.id}/#{attr}" to support id normalization
         [value, result]
       end
 
+      def handle_virtual_attributes(attr, value)
+        case attr
+        when "custom_actions"
+          value = add_custom_action_hrefs(value)
+        when "result_set"
+          value = get_result_set(value)
+        end
+        value
+      end
+
+      def get_result_set(value)
+        offset = request.params[:offset] ? request.params[:offset].to_i : 0
```

Comment:
The ternary operator shouldn't be necessary on these since `nil.to_i` is `0`.

`offset = request.params[:offset].to_i`